app/scripts/modules/amazon/src/instance/amazon.instance.write.service.ts

```diff
@@ -0,0 +1,94 @@
+import { IPromise, module } from 'angular';
+
+import {
+  Application,
+  IMultiInstanceGroup,
+  IMultiInstanceJob,
+  INSTANCE_WRITE_SERVICE,
+  InstanceWriter,
+  ITask,
+  PROVIDER_SERVICE_DELEGATE,
+  ProviderServiceDelegate,
+  SERVER_GROUP_READER,
+  ServerGroupReader,
+  TASK_EXECUTOR,
+  TaskExecutor
+} from '@spinnaker/core';
+
+import { IAmazonInstance } from 'amazon/domain';
+
+export interface IAmazonMultiInstanceGroup extends IMultiInstanceGroup {
+  targetGroups: string[];
+}
+
+export interface IAmazonMultiInstanceJob extends IMultiInstanceJob {
+  targetGroupNames?: string[];
+}
+
+export class AmazonInstanceWriter extends InstanceWriter {
+  public constructor(protected taskExecutor: TaskExecutor,
+                     protected serverGroupReader: ServerGroupReader,
+                     protected providerServiceDelegate: ProviderServiceDelegate) {
+    'ngInject';
+    super(taskExecutor, serverGroupReader, providerServiceDelegate);
+  }
+
+  public deregisterInstancesFromTargetGroup(instanceGroups: IMultiInstanceGroup[], application: Application,
+                                             targetGroupNames: string[]): IPromise<ITask> {
+    const jobs = this.buildMultiInstanceJob(instanceGroups, 'deregisterInstancesFromLoadBalancer') as IAmazonMultiInstanceJob[];
+    jobs.forEach((job) => job.targetGroupNames = targetGroupNames);
+    const descriptor = this.buildMultiInstanceDescriptor(jobs, 'Deregister', `from ${targetGroupNames.join(' and ')}`);
+    return this.taskExecutor.executeTask({
+      job: jobs,
+      application: application,
+      description: descriptor,
+    });
+  }
+
+  public deregisterInstanceFromTargetGroup(instance: IAmazonInstance, application: Application, params: any = {}): IPromise<ITask> {
+    params.type = 'deregisterInstancesFromLoadBalancer';
```

Comment:
Is this right? Does Clouddriver know to look at the `targetGroupNames` to figure out what to do or something?